system/lib/pthread/library_pthread.c

```diff
@@ -636,6 +691,175 @@ int emscripten_sync_run_in_main_runtime_thread_(EM_FUNC_SIGNATURE sig, void *fun
+	emscripten_sync_run_in_main_thread(&q);
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iiiii(int index, int a0, int a1, int a2, int a3)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iifff(int index, int a0, float a1, float a2, float a3)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIFFF;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .f = a1 }, { .f = a2 }, { .f = a3 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iiiiii(int index, int a0, int a1, int a2, int a3, int a4)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 }, { .i = a4 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iiiiiii(int index, int a0, int a1, int a2, int a3, int a4, int a5)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIIIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 }, { .i = a4 }, { .i = a5 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
+EMSCRIPTEN_KEEPALIVE void emscripten_sync_run_in_main_runtime_thread_js_viiiiii(int index, int a0, int a1, int a2, int a3, int a4, int a5)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_VIIIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 }, { .i = a4 }, { .i = a5 } } };
+	emscripten_sync_run_in_main_thread(&q);
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iiiiiiiii(int index, int a0, int a1, int a2, int a3, int a4, int a5, int a6, int a7)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIIIIIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 }, { .i = a4 }, { .i = a5 }, { .i = a6 }, { .i = a7 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
+EMSCRIPTEN_KEEPALIVE int emscripten_sync_run_in_main_runtime_thread_js_iiiiiiiiii(int index, int a0, int a1, int a2, int a3, int a4, int a5, int a6, int a7, int a8)
+{
+	EM_FUNC_SIGNATURE sig = EM_FUNC_SIG_IIIIIIIIII;
+	em_queued_call q = { .functionEnum = sig, .js = index, .args = { { .i = a0 }, { .i = a1 }, { .i = a2 }, { .i = a3 }, { .i = a4 }, { .i = a5 }, { .i = a6 }, { .i = a7 }, { .i = a8 } } };
+	emscripten_sync_run_in_main_thread(&q);
+	return q.returnValue.i;
+}
+
```

Comment:
Perhaps these could be made redundant with the structuring from the previous comment? Having a O(#signatures) number of `EMSCRIPTEN_KEEPALIVE` functions would be quite a hit on code size.